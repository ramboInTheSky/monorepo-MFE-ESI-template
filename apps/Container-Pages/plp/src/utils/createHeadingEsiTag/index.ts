import { urlSanitiser } from "../httpUrlTrimmer"
import env from "../../config/env"

const {REACT_APP_APP_URL} = env

const headingEsiBaseUrl = (SITEURL: string, useDevEsi: boolean) =>
    REACT_APP_APP_URL && useDevEsi ? REACT_APP_APP_URL : SITEURL

export default function createHeadingEsiTag(
    siteUrl: string,
    useDevEsi: boolean,
    url: string,
    totalResults: number,
    title: string,
) {
    const sanitisedUrl = urlSanitiser(url)
    const encodedUrl = encodeURIComponent(sanitisedUrl.replace(siteUrl, ""))

    const esiInclude = `<esi:include src="${headingEsiBaseUrl(
        urlSanitiser(siteUrl),
        useDevEsi,
    )}/seo/seo-headings?urlPath=${encodedUrl}"/>`

    const esiWrapper = `<esi:try>
        <esi:attempt> 
            ${esiInclude}
        </esi:attempt> 
        <esi:except> 
            <h1>${totalResults}&nbsp;${title}</h1>
        </esi:except>
    </esi:try>`

    return esiWrapper
}
