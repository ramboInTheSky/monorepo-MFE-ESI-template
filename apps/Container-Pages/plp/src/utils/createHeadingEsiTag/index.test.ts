import createHeadingEsiTag from "."

jest.mock("../../config/env", () => ({
    REACT_APP_APP_URL: "www.test123.com",
}))

describe(`Given a ${createHeadingEsiTag.name}`, () => {
    describe("When supplied with heading params", () => {
        it("should return an ESI tag", () => {
            const tag = createHeadingEsiTag("www.test.com", true, `/shop/women`, 10, "t-shirts")
            expect(tag).toMatchSnapshot()
        })

        it("When not using the devEsi solution it should return an ESI tag", () => {
            const tag = createHeadingEsiTag("www.test.com", false, "www.test.com/shop/women", 10, "t-shirts")
            expect(tag).toMatchSnapshot()
        })
    })
})
