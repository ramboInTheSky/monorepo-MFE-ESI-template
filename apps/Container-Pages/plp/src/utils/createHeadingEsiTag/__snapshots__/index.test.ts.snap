// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a createHeadingEsiTag When supplied with heading params When not using the devEsi solution it should return an ESI tag 1`] = `
"<esi:try>
        <esi:attempt> 
            <esi:include src=\\"www.test.com/seo/seo-headings?urlPath=%2Fshop%2Fwomen\\"/>
        </esi:attempt> 
        <esi:except> 
            <h1>10&nbsp;t-shirts</h1>
        </esi:except>
    </esi:try>"
`;

exports[`Given a createHeadingEsiTag When supplied with heading params should return an ESI tag 1`] = `
"<esi:try>
        <esi:attempt> 
            <esi:include src=\\"www.test123.com/seo/seo-headings?urlPath=%2Fshop%2Fwomen\\"/>
        </esi:attempt> 
        <esi:except> 
            <h1>10&nbsp;t-shirts</h1>
        </esi:except>
    </esi:try>"
`;
